### PR TITLE
feat(ui): dialog component without style

### PR DESCRIPTION
**Pull request by DavidLHW**
This change is done in part for the `Combobox` component that is referencing [Shadcn's `Combobox`](https://ui.shadcn.com/docs/components/combobox) component.

Styling will be coordinated in a separate PR with the implementation of the mobile screen and styles

## changes

- add `@radix-ui/react-dialog` package
- add `Dialog` component, referenced from [Shadcn's `Dialog`](https://ui.shadcn.com/docs/components/dialog)
- add subcomponents `DialogHeader`, `DialogFooter`, `DialogTitle`, `DialogDescription`
- add close icon X
- add usage example at `/components`

## preview

image

### CODE REVIEW

File: src/common/components/DialogDescription/DialogDescription.theme.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+export type DialogDescriptionVariants = VariantProps<
+  typeof dialogDescriptionTheme
+>;
+
+export const dialogDescriptionTheme = tv(
+  {
+    slots: {
+      dialogDescription: ["text-muted-foreground", "text-sm"],
+    },
```
Comment (Aztriltus) [inline src/common/components/DialogDescription/DialogDescription.theme.ts:11]: Should we just directly use the base key?

```suggestion
    base: ["text-muted-foreground", "text-sm"],
```

Comment (Aztriltus) [reply src/common/components/DialogDescription/DialogDescription.theme.ts:11]: Same goes for all the Dialog composition components. Wdyt?

**Review by Aztriltus — CHANGES_REQUESTED** [review]
Some comments!

File: src/common/components/DialogDescription/DialogDescription.theme.ts (hunk 1 above)
Comment (DavidLHW) [reply src/common/components/DialogDescription/DialogDescription.theme.ts:11]: sure! will add them in

File: src/common/components/Dialog/Dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+export const DialogClose = DialogPrimitive.Close;
+
+export type DialogProps = DialogVariants & {
+  trigger: ReactNode;
+  children: ReactNode;
+  dialogRootProps?: ComponentPropsWithoutRef<typeof DialogRoot>;
+  dialogOverlayProps?: ComponentPropsWithoutRef<typeof DialogOverlay>;
+  dialogTriggerProps?: ComponentPropsWithoutRef<typeof DialogTrigger>;
+  dialogContentProps?: ComponentPropsWithoutRef<typeof DialogPrimitive.Content>;
+  dialogCloseProps?: ComponentPropsWithoutRef<typeof DialogClose>;
```
Comment (Aztriltus) [inline src/common/components/Dialog/Dialog.tsx:37]: Similar to Popover PR comment, shall we let the consumer add the composition components like trigger and content as children components in Dialog?

**Review by Aztriltus — CHANGES_REQUESTED** [review]
Some comments. Thanks for updating the theme!

File: src/common/components/Dialog/Dialog.tsx (hunk 2 above)
Comment (DavidLHW) [reply src/common/components/Dialog/Dialog.tsx]: sure! updated at b61142f

**Review by Aztriltus — APPROVED** [review]
LGTM, thanks for the changes 🙌

**Pull request merged**